examples/with-cookie-auth/utils/auth.js

```diff
@@ -0,0 +1,82 @@
+import { Component } from 'react'
+import Router from 'next/router'
+import nextCookie from 'next-cookies'
+import cookie from 'js-cookie'
+import fetch from 'isomorphic-unfetch'
+
+export const login = async ({ username }) => {
+  try {
+    const url = 'https://with-cookie-api.now.sh/login'
+    const response = await fetch(url, {
+      method: 'POST',
+      headers: { 'Content-Type': 'application/json' },
+      body: JSON.stringify({ username })
+    })
+    if (response.ok) {
+      const { token } = await response.json()
+      cookie.set('token', token, { expires: 1 })
+      Router.push('/profile')
+    } else {
+      console.log('Login failed.')
+      // https://github.com/developit/unfetch#caveats
+      let error = new Error(response.statusText)
+      error.response = response
+      return Promise.reject(error)
+    }
+  } catch (error) {
+    console.error('You have an error in your code or there are Network issues.', error)
+    throw new Error(error)
+  }
+}
+
+export const logout = () => {
+  cookie.remove('token')
+  // to support logging out from all windows
+  window.localStorage.setItem('logout', Date.now())
```

Comment:
That's an interesting trick! I need to remember it 😄

Reply:
When a user logs out, they assume all of their data has been removed. Evidence that they have visited the site before isn't removing all data. Feel like this could have GDPR/Privacy Policy implications.

Reply:
Yeah, the use of `Date.now` keeps the date they logged out. Probably an empty object could do the trick, I need to put something there so it can trigger the event.

Other solution could be a global event emitter, but I don't think is a good idea.

I'll try with an empty object or a "random" string.

Reply:
Oh ok then why not call it "isLoggingOut" and set it to `true` or `1`.

Seems borderline hacky but I like the innovation